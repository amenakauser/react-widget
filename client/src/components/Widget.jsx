import React from 'react';
import { Row, Col, Button, Icon } from 'antd';
import { Transition } from 'react-transition-group';
import 'antd/dist/antd.css';
import '../../styles/widget-style.css'

class Widget extends React.Component {
	constructor(props){
		super(props);
		this.state = {
			open : false,
			showDockedWidget : true
		}
	}

	handleToggleOpen = () => {
		this.setState((prev) => {
			let { showDockedWidget } = prev;
			if (!prev.open) {
				showDockedWidget = false;
			}
			return {
				showDockedWidget,
				open: !prev.open
			}
		})
	}

	handleWidgetExit = () => {
		this.setState({
			showDockedWidget: true
		})
	}

	renderBody = () => {
		if (this.state.showDockedWidget) {
			return (
				<Icon 
					type="file-text" 
					style={{ fontSize: 38, color: '#08c', borderRadius: '10px', backgroundColor: '#0088cc5e', padding: '15px'}} 
					className="dock-button" 
					onClick={this.handleToggleOpen} />
				);
		}
		return '';
	}

	render() {
		const duration = 250;
		const defaultStyle = {
  				transition: `opacity ${duration}ms ease-in-out`,
  				opacity: 0,
			}

		const transitionStyles = {
  				entering: { opacity: 0 },
  				entered:  { opacity: 1 },
			}
		const body = this.renderBody();
		return (
			<div className="docked-widget">
			<Transition 
				in={this.state.open} 
				timeout={duration}
				onExited={this.handleWidgetExit}>
				 {status => (
				 <div 
				 style={{...defaultStyle, ...transitionStyles[status]}} 
				 className={`widget widget-${status}`}>
					<Row className="widget-dialog">
					<Col span={18} className="widget-title">Hi Sravanthi! </Col>
					<Col span={6} className="widget-header-icon" onClick={this.handleToggleOpen}>X</Col>
					<span className="company-title">Salesforce Knowledge base</span>
					</Row>
					<Row className="widget-body">
					<Col className="body-title">Advice and anssers from Salesforce</Col>
					</Row>
					<Row className="widget-footer">
					Footer
					</Row>
				</div>
				)}
			</Transition>
			{body}
			</div>
			)
	}
}
export default Widget;