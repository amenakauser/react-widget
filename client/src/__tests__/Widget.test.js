import React from 'react';
import { shallow } from 'enzyme';
import Widget from '../components/Widget.jsx';

describe('<Widget/>', () => {
  const mockFn = jest.fn();
  it('should be defined', () => {
    expect(Widget).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = shallow(
      <Widget />
    );
    expect(tree).toMatchSnapshot();
  });
});

