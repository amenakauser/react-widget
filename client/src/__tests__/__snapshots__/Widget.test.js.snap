// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Widget/> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Widget />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Transition
          appear={false}
          enter={true}
          exit={true}
          in={false}
          mountOnEnter={false}
          onEnter={[Function]}
          onEntered={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onExiting={[Function]}
          timeout={250}
          unmountOnExit={false}
        >
          [Function]
        </Transition>,
        <Icon
          className="dock-button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#159adc",
              "borderRadius": "50%",
              "color": "#FFF",
              "fontSize": 38,
              "padding": "15px",
            }
          }
          type="book"
        />,
      ],
      "className": "docked-widget",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "appear": false,
          "children": [Function],
          "enter": true,
          "exit": true,
          "in": false,
          "mountOnEnter": false,
          "onEnter": [Function],
          "onEntered": [Function],
          "onEntering": [Function],
          "onExit": [Function],
          "onExited": [Function],
          "onExiting": [Function],
          "timeout": 250,
          "unmountOnExit": false,
        },
        "ref": null,
        "rendered": [Function],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "dock-button",
          "onClick": [Function],
          "style": Object {
            "backgroundColor": "#159adc",
            "borderRadius": "50%",
            "color": "#FFF",
            "fontSize": 38,
            "padding": "15px",
          },
          "type": "book",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Transition
            appear={false}
            enter={true}
            exit={true}
            in={false}
            mountOnEnter={false}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            timeout={250}
            unmountOnExit={false}
          >
            [Function]
          </Transition>,
          <Icon
            className="dock-button"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#159adc",
                "borderRadius": "50%",
                "color": "#FFF",
                "fontSize": 38,
                "padding": "15px",
              }
            }
            type="book"
          />,
        ],
        "className": "docked-widget",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "appear": false,
            "children": [Function],
            "enter": true,
            "exit": true,
            "in": false,
            "mountOnEnter": false,
            "onEnter": [Function],
            "onEntered": [Function],
            "onEntering": [Function],
            "onExit": [Function],
            "onExited": [Function],
            "onExiting": [Function],
            "timeout": 250,
            "unmountOnExit": false,
          },
          "ref": null,
          "rendered": [Function],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "dock-button",
            "onClick": [Function],
            "style": Object {
              "backgroundColor": "#159adc",
              "borderRadius": "50%",
              "color": "#FFF",
              "fontSize": 38,
              "padding": "15px",
            },
            "type": "book",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
